linetools/isgm/io.py

```diff
@@ -41,3 +51,83 @@ def abssys_from_json(filename):
 
     # Return
     return abs_sys
+
+
+def read_joebvp(filename, coord, llist=None, specfile=None, chk_vel=False):
+    """
+    Parameters
+    ----------
+    filename : str
+      joeB VP filename
+    coord : SkyCoord
+    llist : LineList, optional
+    specfile : str, optional
+    chk_vel : bool, optional
+      Demand that the velocities of a given ion all be the same
+
+    Returns
+    -------
+    comps : list
+      list of AbsComponent objects
+    """
+    # init
+    if llist is None:
+        llist = LineList('ISM')
+    comps = []
+    # Read
+    vp_data = Table.read(filename, format='ascii')
+
+    # Subset by zsys + trans
+    lbls = []
+    for izsys, itrans in zip(vp_data['zsys'], vp_data['trans']):
+        lbls.append('{:.6f}_{:s}'.format(izsys, itrans))
+    lbls = np.array(lbls)
+    ulbls = np.unique(lbls)
+
+    # Subset by nflag; Build components
+    for lbl in ulbls:
+        mt_lines = np.where(lbls == lbl)[0]
+        if chk_vel:
+            if len(np.unique(vp_data['vel'][mt_lines])) != 1:
+                pdb.set_trace()
+        z_fit = vp_data['zsys'][mt_lines[0]] + vp_data['vel'][mt_lines[0]] * (1 + vp_data['zsys'][mt_lines[0]]) / ckms
+        # Loop on abs lines
+        alines = []
+        for idx in mt_lines:
+            zlim = [vp_data['zsys'][idx] +
+                    vp_data[vkey][idx] * (1 + vp_data['zsys'][idx]) / ckms
+                    for vkey in ['vlim1', 'vlim2']]
+            absline = AbsLine(vp_data['restwave'][idx] * u.AA, z=z_fit,
+                              zlim=zlim, linelist=llist)
+            # Add measurements [JB -- Want to capture anything else??]
+            absline.attrib['coord'] = coord
+            absline.attrib['flag_N'] = 1
+            absline.attrib['logN'] = vp_data['col'][idx]
+            absline.attrib['sig_logN'] = vp_data['sigcol'][idx]
+            absline.attrib['b'] = vp_data['bval'][idx]
+            absline.attrib['sig_b'] = vp_data['sigbval'][idx]
+            absline.attrib['z'] = z_fit
+            absline.attrib['sig_z'] = (1 + vp_data['z_comp'][idx]) * vp_data['sigvel'][idx] / ckms
+            if specfile is None:
+                absline.attrib['specfile'] = vp_data['specfile'][idx]
```

Comment:
Is this something we want? If the joebvp file already has the specfile, why should we allow the user to change it?

Reply:
@jnburchett had a good reason (see above)